File: php/src/Advertisements/Advertisement/UI/Http/PublishAdvertisementController.php
<?php
declare(strict_types=1);

namespace Demo\App\Advertisements\Advertisement\UI\Http;

use Demo\App\Advertisements\Advertisement\Application\Command\PublishAdvertisement\PublishAdvertisementCommand;
use Demo\App\Advertisements\Advertisement\Application\Command\PublishAdvertisement\PublishAdvertisementUseCase;
use Demo\App\Common\Exceptions\BoundedContextException;
use Demo\App\Common\UI\CommonController;
use Demo\App\Framework\FrameworkRequest;
use Demo\App\Framework\FrameworkResponse;
use Demo\App\Framework\SecurityUser\FrameworkSecurityService;

final class PublishAdvertisementController extends CommonController
{
    public function __construct(
        private PublishAdvertisementUseCase $useCase,
        private FrameworkSecurityService $securityService,
    ) {}

    public function request(FrameworkRequest $request, array $pathValues = []): FrameworkResponse
    {
        try {
            $user = $this->securityService->getSecurityUserFromRequest($request);

            if (null == $user || !$user->role() == 'member') {
                return $this->processUnauthorizedResponse();
            }

            $command = new PublishAdvertisementCommand(
                $user->id(),
                $user->role(),
                ($request->content())['id'],
                ($request->content())['description'],
                ($request->content())['email'],
                ($request->content())['password'],
                ($request->content())['memberId'],
                ($request->content())['civicCenterId'],
            );

            $this->useCase->execute($command);

            return $this->processSuccessfulCreateCommand();
        } catch (BoundedContextException $exception) {
            return $this->processDomainOrApplicationExceptionResponse($exception);
        } catch (\Throwable $exception) {
            return $this->processGenericException($exception);
        }
    }
}

File: kotlin/src/main/kotlin/advertisements/advertisement/ui/http/PublishAdvertisementController.kt
package advertisements.advertisement.ui.http

import advertisements.advertisement.application.command.publishAdvertisement.PublishAdvertisementCommand
import advertisements.advertisement.application.command.publishAdvertisement.PublishAdvertisementUseCase
import common.exceptions.BoundedContextException
import common.ui.http.CommonController
import framework.FrameworkRequest
import framework.FrameworkResponse
import framework.securityuser.FrameworkSecurityService

class PublishAdvertisementController(
    private val useCase: PublishAdvertisementUseCase,
    private val securityService: FrameworkSecurityService,
): CommonController(){

    override fun execute(request: FrameworkRequest, pathValues: Map<String, String>): FrameworkResponse {
        try {
            val user = securityService.getSecurityUserFromRequest(request)

            if (user?.role != "member") {
                return processUnauthorizedResponse()
            }

            useCase.execute(
                PublishAdvertisementCommand(
                    user.id,
                    user.role,
                    request.content["id"]!!,
                    request.content["description"]!!,
                    request.content["email"]!!,
                    request.content["password"]!!,
                    request.content["memberId"]!!,
                    request.content["civicCenterId"]!!
                )
            )

            return processSuccessfulCreateCommand()
        } catch (e: BoundedContextException) {
            return processApplicationOrDomainException(e)
        } catch (e: Exception) {
            return processGenericException(e)
        }
    }
}
